analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs

```diff
@@ -41,14 +43,35 @@ private class TestSetup
         {
             public string Path { get; set; }
             public SonarDiagnosticAnalyzer Analyzer { get; set; }
+            public IEnumerable<MetadataReference> AdditionalReferences { get; set; } = new MetadataReference[] { };
         }
 
+        // Various classes that invoke all the `ReportDiagnosticWhenActive` methods in AnalysisContextExtensions
+        // We mention in comments the type of Context that is used to invoke (directly or indirectly) the `ReportDiagnosticWhenActive` method
         private readonly List<TestSetup> testCases = new List<TestSetup>(new[]
```

Comment:
I don't really like how this test is structured, I'll think about how to refactor it using `[DataTestMethod]`

Reply:
That won't let you use dynamic stuff like those references :/